packages/@aws-cdk/core/README.md

```diff
@@ -15,8 +15,236 @@
+
+### Mappings
+
+CloudFormation [mappings][cfn-mappings] are created and queried using the
+`CfnMappings` class:
+
+```ts
+const mapping = new CfnMapping(this, 'MappingTable', {
+  mapping: {
+    regionName: {
+      'us-east-1': 'US East (N. Virginia)',
+      'us-east-2': 'US East (Ohio)',
+      // ...
+    },
+    // ...
+  }
+});
+
+mapping.findInMap('regionName', Aws.REGION);
+```
+
+[cfn-mappings]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/mappings-section-structure.html
+
+### Stack Outputs
+
+CloudFormation [stack outputs][cfn-stack-output] and exports are created using
+the `CfnOutput` class:
+
+```ts
+new CfnOutput(this, 'OutputName', {
+  value: bucket.bucketName,
+  description: 'The name of an S3 bucket', // Optional
+  exportName: 'Global.BucketName', // Registers a CloudFormation export
+});
+```
+
+[cfn-stack-output]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/outputs-section-structure.html
+
+### Template Options & Transform
+
+CloudFormation templates support a number of options, including which Macros or
+[Transforms][cfn-transform] to use when deploying the stack. Those can be
+configured using the `stack.templateOptions` property:
+
+```ts
+const stack = new Stack(app, 'StackName');
+
+stack.templateOptions.description = 'This will appear in the AWS console';
+stack.templateOptions.transform = 'AWS::Serverless';
+stack.templateOptions.metadata = {
+  metadataKey: 'MetadataValue',
+};
+```
+
+[cfn-transform]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/transform-section-structure.html
+
+### Resource Options
+
+Similar to template options, CloudFormation resources can also specify
+[resource attributes][cfn-resource-attributes]. The `CfnResource` class allows
+accessing those though the `cfnOptions` property:
+
+```ts
+const rawBucket = new s3.CfnBucket(this, 'Bucket', { /* ... */ });
+rawBucket.condition = new CfnCondition(this, 'EnableBucket', { /* ... */ });
+rawBucket.cfnOptions.metadata = {
+  metadataKey: 'MetadataValue',
+};
+```
+
+Resource dependencies (the `DependsOn` attribute) is modified using the
+`cfnResource.addDependsOn` method:
+
+```ts
+const resourceA = new CfnResource(this, 'ResourceA', { /* ... */ });
+const resourceB = new CfnResource(this, 'ResourceB', { /* ... */ });
+
+resourceB.addDependsOn(resourceA);
+```
+
+[cfn-resource-attributes]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-product-attribute-reference.html
+
+### Emitting Raw Resources
+
+The `CfnResource` class allows emitting arbitrary entries in the
+[Resources][cfn-resources] section of the CloudFormation template.
+
+```ts
+new CfnResource(this, 'ResourceId', {
+  type: 'AWS::S3::Bucket',
+  properties: {
+    BucketName: 'bucket-name'
+  },
+});
+```
+
+As for any other resource, the logical ID in the CloudFormation template will be
+generated by the AWS CDK, but the type and properties will be copied verbatim in
+the synthesized template.
 
```

Comment:
`CfnInclude` is missing and I think there's something else (look at all the CfnXxx classes in core)